Sources/Basics/Triple+Basics.swift

```diff
@@ -146,7 +146,9 @@ extension Triple {
         case .wasi:
             return ".wasm"
         default:
-            fatalError("Cannot create dynamic libraries for os \"\(os)\".")
+            // Fallback to a nonsense extension for operating systems with
+            // unknown conventions.
+            return ".dynamiclibrary"
```

Comment:
Any reason this is preferred over `.so`?

Reply:
I don't really like the idea of the extension for "macho" being "so". I think that would be kinda surprising to the user. 

I was considering this default branch determining the extension based on the object file format instead, thought?

Reply:
What is the behavior if building on a non-macho platform like Linux? Would this still work?

Reply:
This code path is only used to plan the build of a dynamic library product for an os none triple. Actually doing this build is nonsensical because the dynamic library would require a specific dynamic loader provided by the OS and by definition, os none does not have a dynamic loader.

One alternative option is to use the object format e.g.:
```swift
switch os {
case _ where isDarwin():
    return ".dylib"
case .linux, .openbsd:
    return ".so"
case .win32:
    return ".dll"
case .wasi:
    return ".wasm"
default:
  break
}

guard let objectFormat = self.objectFormat else {
    fatalError("Cannot create dynamic libraries unknown object format.")
}

switch objectFormat {
case .coff:
    return ".coff"
case .elf:
    return ".elf"
case .macho:
    return ".macho"
case .wasm:
    return ".wasm"
case .xcoff:
    return ".xcoff"
}
```

Reply:
So this basically just spoofs the "dynamic library" on a none OS but doesn't actually generate anything? That's kind of weird. But if that's the case then I guess it's just arbitrary what you pick there.

Reply:
yeah its not a great thing to do, but fixing the underlying issue with build planning is something the swift pm is working on

Reply:
Well, I'm okay with this if it fixes the none OS issue and if it doesn't break anything!

Reply:
I think doing this based on object format looks better.